lib/rex/text.rb

```diff
@@ -1522,7 +1522,8 @@ def self.badchar_index(data, badchars = '')
   # @param data [#delete]
   # @param badchars [String] A list of characters considered to be bad
   def self.remove_badchars(data, badchars = '')
-    data.delete(badchars)
+    badchars_pat = badchars.unpack("C*").map{|c| "\\x%.2x" % c}.join
+    data.gsub!(/[#{badchars_pat}]/n, '')
```

Comment:
I guess no caller depended on the in-place modification or it wouldn't have worked at all. The current code (last commit) should handle both cases at least.